### PR TITLE
Adding confirmation modal when setting the board to public with Editor or Commenter role

### LINKED ISSUE #3863: Bug: Missing minimum role confirmation modal when switching from private to public board
## Steps to reproduce the behavior

1. Go to a private board
2. Add a few explicit board members and set their roles to Viewer or Commenter
3. Switch team role to Editor
4. See error - confirmation modal does not appear and board is switched to public

## Expected behavior

Confirmation modal appears to inform admin that some explicit board members will get their roles promoted. See designs:

image

#### Current behavior:
https://user-images.githubusercontent.com/93531870/190785743-63a12434-e629-403c-838f-d773adb3ee05.mov

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: v7.3 Bug Bash
 - Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
I added a modal to ensure that you confirm that you want to set the role
generating "promotion" of users. That only can happen on Editor and Commenter,
so I only show that in that cases.

#### Ticket Link
Fixes #3863

### CODE REVIEW

**Review by sbishel — COMMENTED** [review]
I got the modal anytime the team permissions changed from/to Editor and/or commenter. I'm not sure if this is correct or not. The ticket says when making the board public. So that implies to display when going None -> Editor, or None -> Commenter. However, the message mentions increasing permissions, so I can see mentioning whenever that change is made. Probably need some input from @wuwinson 

https://user-images.githubusercontent.com/12704875/193668255-09093602-ccbd-4392-b342-4155f63a3f58.mov

**Comment by wuwinson** [comment]
@sbishel @jespino Good point. I missed the use case of upgrading users on existing public boards. The current implementation is better than what's on the spec, but the messaging on the confirmation modal is now misleading/confusing. Is there enough time to change the copy? If not, maybe we ship v7.4 without the modal and add it in for v7.5.

Proposed copy changes for the modal:
>**Change minimum board role**
>Everyone on the board that has a lower permission than the "[selected role]" role will **now be promoted to [selected role]**. Are you sure you want to change the minimum role for the board?
>`Cancel` `Change board role`

cc @justinegeffen @asaadmahmood for review.

**Comment by jespino** [comment]
@wuwinson done

**Review by justinegeffen — COMMENTED** [review]
I've added some comments around wording and terminology.

File: webapp/i18n/en.json [hunk 1]
```diff
@@ -401,6 +401,9 @@
   "share-board.publish": "Publish",
   "share-board.share": "Share",
   "shareBoard.channels-select-group": "Channels",
+  "shareBoard.confirm-change-team-role.body": "Change board role",
+  "shareBoard.confirm-change-team-role.confirmBtnText": "Everyone on the board that has a lower permission than the \"{role}\" role will <b>now be promoted to {role}</b>. Are you sure you want to change the minimum role for the board?",
+  "shareBoard.confirm-change-team-role.title": "Change minimum board role",
```
[the remark this answers is not in the document]
Comment (wuwinson) [reply webapp/i18n/en.json:406]: @justinegeffen This is part of the new feature we're releasing in v7.3/7.4. It is the lowest role you will be able to set for individual board members.

File: webapp/i18n/en.json [hunk 2]
```diff
@@ -401,6 +401,9 @@
   "share-board.publish": "Publish",
   "share-board.share": "Share",
   "shareBoard.channels-select-group": "Channels",
+  "shareBoard.confirm-change-team-role.body": "Change board role",
+  "shareBoard.confirm-change-team-role.confirmBtnText": "Everyone on the board that has a lower permission than the \"{role}\" role will <b>now be promoted to {role}</b>. Are you sure you want to change the minimum role for the board?",
```
[the remark this answers is not in the document]
Comment (wuwinson) [reply webapp/i18n/en.json:405]: @justinegeffen Only board members who have a lesser role will be promoted to the role, thus the minimum board role. Anyone else who has a role that is equal to or higher than the min role will not be impacted.

File: webapp/i18n/en.json [hunk 3]
```diff
@@ -401,6 +401,9 @@
   "share-board.publish": "Publish",
   "share-board.share": "Share",
   "shareBoard.channels-select-group": "Channels",
+  "shareBoard.confirm-change-team-role.body": "Change board role",
```
Comment (justinegeffen) [inline webapp/i18n/en.json:404]:
```suggestion
  "shareBoard.confirm-change-team-role.body": "Change minimum board role",
```

File: webapp/i18n/en.json (hunk 2 above)
Comment (justinegeffen) [inline webapp/i18n/en.json:405]:
```suggestion
  "shareBoard.confirm-change-team-role.confirmBtnText": "Everyone on this board with a lower permission than the  \"{role}\" role will <b>now be promoted to {role}</b>. Are you sure you want to change the minimum role for the board?",
```

File: webapp/src/components/shareBoard/teamPermissionsRow.tsx [hunk 4]
```diff
@@ -53,8 +62,33 @@ const TeamPermissionsRow = (): JSX.Element => {
...
+    const confirmationDialog = (
+        <ConfirmationDialogBox
+            dialogBox={{
+                heading: intl.formatMessage({
+                    id: 'shareBoard.confirm-change-team-role.title',
+                    defaultMessage: 'Change minimum board role',
+                }),
+                subText: intl.formatMessage({
+                    id: 'shareBoard.confirm-change-team-role.body',
+                    defaultMessage: 'Everyone on the board that has a lower permission than the "{role}" role will <b>now be promoted to {role}</b>. Are you sure you want to change the minimum role for the board?',
```
Comment (justinegeffen) [inline webapp/src/components/shareBoard/teamPermissionsRow.tsx:74]:
```suggestion
                    defaultMessage: 'Everyone on this board with a lower permission than the "{role}" role will <b>now be promoted to {role}</b>. Are you sure you want to change the minimum role for the board?',
```

**Review by Pinjasaur — APPROVED** [review]

**Pull request merged**